### PR TITLE
check python interpreter by --version option

### PR DESCRIPTION
**Pull request by namuyan**
https://github.com/PyO3/pyo3/issues/619
solved on my condition (Windows10 64bit, Python 3.6.7 :: Anaconda custom (64-bit))

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -25,10 +25,13 @@ const MIN_VERSION: &'static str = "1.37.0-nightly";
 
 lazy_static! {
     static ref PYTHON_INTERPRETER: &'static str = {
-        ["python3", "python"]
+        ["python", "python3"]
             .iter()
-            .map(|bin| (bin, Command::new(bin).spawn()))
-            .find(|(_, r)| r.is_ok())
+            .map(|bin| (bin, Command::new(bin).arg("--version").output()))
```
Comment (kngwyu) [inline build.rs]: How about this:
```rust
            .find(|bin| {
                if let Ok(out) = Command::new(bin).arg("--version").output() {
                    out.stdout.starts_with(b"Python 3")
                } else {
                    false
                }
            })
            .expect("Python 3.x interpreter not found")
```

**Comment by kngwyu** [comment]
Thank you for fixing this!
I left a comment about coding, but if you'd prefer the current code, I merge this PR as is.

File: build.rs (hunk 1 above)
Comment (namuyan) [reply build.rs]: Thank you for comment!
Travis show me one curious things, Windows Anaconda3 Python output version by stderr. Python provided by official output version by stdout, only Anaconda3.
So It looks Anaconda3 bug. No one noticed.

Comment (kngwyu) [reply build.rs]: Ah, OK, thank you for investigating that.
My intention is just to shorten codes, though.

Comment (namuyan) [reply build.rs]: No problem, I feel method chain is not readable code.
I changed to something like your suggestion.

**Comment by kngwyu** [comment]
Thanks!

**Pull request merged**